code/__DEFINES/integrated_electronics.dm

```diff
@@ -46,3 +46,6 @@
 
 // Data limits.
 #define IC_MAX_LIST_LENGTH			500
+
+// Type checks
+#define CIRCUIT_VALID_ATOM(_CHECKING_ATOM)	(!(ismob(_CHECKING_ATOM) && !isliving(_CHECKING_ATOM)))
```

Comment:
objs?turfs?Even if you insist on ghost detection removal, make proc in atom circuit_interaction_check.Anyway, some objs will be blacklisted later for detection.

Reply:
the inner statement returns true if the object is a mob and not a living mob
the outer statement returns true if the inner statement doesn't
therefore it is true if the object is not a non /mob/living mob. this is to avoid proccall overhead.

Reply:
stop defining this kind of shit